### DISCUSSION TITLE
Not seeing any HPA after deploying scaledobject

### DISCUSSION THREAD

**Discussion by Savitha-Amaldev** [kedacore/keda#1858] [Q&A / Need Help]
After deploying scaled objects, I'm not able to verify whether it started scaling or not. when I describe the scaled object, I'm not seeing any events triggered. Before I used to see events as scaled object is active or if there is any warning or not How to check whether something is missing in my environment or not. Any help for debugging the scaled objects are greatly appreciated

**Comment by Savitha-Amaldev** [comment]
my scaled object looks like this:

---
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: prometheus-keda-scaledobject
  namespace: aa
spec:
  scaleTargetRef:
    name: fastload-lcache-mem-only
  pollingInterval: 60 # Optional. Default: 30 seconds
  minReplicaCount: 2 # Optional. Default: 0
  maxReplicaCount: 20 # Optional. Default: 100
  triggers:
    - type: prometheus
      metadata:
       metricName: nginx_ingress_controller_requests
       query: nginx_ingress_controller_requests{job="metrics-ingress",namespace="aa", service="xyz"}
       serverAddress: "http://localhost:9090"
       threshold: "50"

**Comment by tomkerkhove** [comment]
Can you please provide the KEDA logs?

**Comment by tomkerkhove** [comment]
And what is the output of kubectl get scaledobjects?

**Comment by Savitha-Amaldev** [comment]
> Can you please provide the KEDA logs?

I couldn't find much logs regarding this.. I used "kubectl logs scaledobjects/scaledobjectname -n scaledobject_deployed_namespace  -c  keda-operator" command and got a one line error which I'm mentioning below
error: no kind "ScaledObject" is registered for version "keda.sh/v1alpha1" in scheme "k8s.io/kubectl/pkg/scheme/scheme.go:28"

**Comment by Savitha-Amaldev** [comment]
> And what is the output of kubectl get scaledobjects?

I can see my scaledobject in the namespace

**Comment by tomkerkhove** [comment]
Our CRD doesn't provide any logs, you need to check the operator of KEDA itself, can you check the logs there and share the output please?

Also, can you run `kubectl get scaledobject` and share the output please?

**Comment by zroubalik** [comment]
And I would add `kubectl get crd | grep keda`, seems like your CRDs are not correctly installed

**Comment by Savitha-Amaldev** [comment]
> Our CRD doesn't provide any logs, you need to check the operator of KEDA itself, can you check the logs there and share the output please?
> 
> Also, can you run `kubectl get scaledobject` and share the output please?
![Scaledobjects](https://user-images.githubusercontent.com/66904054/120619383-c5ae7e00-c410-11eb-94a1-7de58120b2bf.PNG)
              
This is  how it looks.. crds are installed correctly as i can see one hpa created for one scaledobject. But none of the others are getting created

**Comment by tomkerkhove** [comment]
Can you also share the logs of the KEDA operator please?

**Reply by Savitha-Amaldev** [reply]
> Can you also share the logs of the KEDA operator please?

  > Can you also share the logs of the KEDA operator please?

 1 status.go:71] apiserver received an error that is not an metav1.Status: &errors.errorString{s:"No matching metrics found for prometheus-http---localhost-9090-nginx_ingress_controller_requests"}
E0603 04:24:12.796081       1 prometheus_scaler.go:187] keda_metrics_adapter/prometheus_scaler "msg"="error executing prometheus query" "error"="Get \"http://localhost:9090/api/v1/query?query=nginx_ingress_controller_requests%7Bjob%3D%22metrics-ingress%22%2Cnamespace%3D%22cpqa5%22%2C+service%3D%22xyzxyzxyzxyzxyzxyzxyz%22%7D&time=2021-06-03T04:24:12Z\": dial tcp [::1]:9090: connect: connection refused"
E0603 04:24:12.796138       1 provider.go:99] keda_metrics_adapter/provider "msg"="error getting metric for scaler" "error"="Get \"http://localhost:9090/api/v1/query?query=nginx_ingress_controller_requests%7Bjob%3D%22metrics-ingress%22%2Cnamespace%3D%22cpqa5%22%2C+service%3D%22xyzxyzxyzxyzxyzxyzxyz%22%7D&time=2021-06-03T04:24:12Z\": dial tcp [::1]:9090: connect: connection refused"  "scaledObject.Name"="prometheus-keda-scaledobject" "scaledObject.Namespace"="xxx5" "scaler"={}

**Reply by Savitha-Amaldev** [reply]
This is what Im seeing in operator metrics log and  operator logs are as follows : 

2021-06-03T09:15:48.260Z        ERROR   prometheus_scaler       error executing prometheus query        {"error": "Get \"http://localhost:9090/api/v1/query?query=nginx_ingress_controller_requests%7Bjob%3D%22metrics-ingress%22%2Cnamespace%3D%22cpqa5%22%2C+service%3D%22xyzxyzxyzxyzxyzxyzxyz%22%7D&time=2021-06-03T09:15:48Z\": dial tcp [::1]:9090: connect: connection refused"}
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScaledObjectScalers
        /workspace/pkg/scaling/scale_handler.go:216
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
        /workspace/pkg/scaling/scale_handler.go:205
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
        /workspace/pkg/scaling/scale_handler.go:146

**Reply by tomkerkhove** [reply]
Oh I have not noticed your server address which is pointing to localhost, you'll need to fix that since KEDA is unable to connect to it (hence logs above)

**Reply by Savitha-Amaldev** [reply]
I changed it to serverip by editing the scaledobject script.. But no luck.. It's not reflecting in logs as well and if i'm deploying new scaledobject with serverip instead of localhost, no hpa is getting created. Usually how long does it takes to get a hpa activated?

**Reply by tomkerkhove** [reply]
If it can connect to Prometheus, it will create the HPA if it has to do 1<->n scaling.

**Reply by Savitha-Amaldev** [reply]
Any idea why it's not happening here? No hpa is getting created for mssql trigger as well.. I'm not getting any logs for that either
